### PR TITLE
Added company and completed some models

### PR DESCRIPTION
**Pull request by datanel**
- Added Company (related to https://github.com/CanalTP/navitia/pull/2282)
- Updated PhysicalMode
- Updated Route
- Added NaiveDate for dates fix https://github.com/CanalTP/navitia_model/issues/17

### CODE REVIEW

File: src/ntfs.rs [hunk 1]
```diff
@@ -105,6 +105,8 @@ struct StopTime {
     #[serde(default, deserialize_with = "de_from_u8", serialize_with = "ser_from_bool")]
     datetime_estimated: bool,
     local_zone_id: Option<u16>,
+    headsign: Option<String>,
+    stop_time_id: Option<String>,
```
Comment (TeXitoi) [inline src/ntfs.rs]: StopTime is the object that use the RAM. Thus, what's the impact of this change on fr-idf?

We may need to manage these data from outside (with something like a `stop_time_ids: HashMap<(Idx<VehicleJourney>, u32), String>`) as these fields are rarely used.

Comment (datanel) [reply src/ntfs.rs]: without stop_time_id and headsign
1.03 GB

without stop_time_id
1.40 GB

with stop_time_id and headsign
2.17 GB :open_mouth:

I remove these 2 lines, I will do something better as you suggest in another PR

Comment (TeXitoi) [reply src/ntfs.rs]: That's almost always None, thus an external storage will be less costy

Comment (prhod) [reply src/ntfs.rs]: It's not very used for the moment, but I think it will be more and more used in the future.

File: src/objects.rs [hunk 2]
```diff
@@ -376,6 +380,25 @@ impl Id<Calendar> for Calendar {
...
+pub struct Company {
+    #[serde(rename = "company_id")] pub id: String,
+    #[serde(rename = "company_name")] pub name: String,
+    #[serde(rename = "company_address_name")] pub address_name: Option<String>,
+    #[serde(rename = "company_address_number")] pub address_number: Option<String>,
+    #[serde(rename = "company_address_type")] pub address_type: Option<String>,
+    #[serde(rename = "company_url")] pub url: Option<String>,
+    #[serde(rename = "company_mail")] pub mail: Option<String>,
+    #[serde(rename = "company_phone")] pub phone: Option<String>,
+    #[serde(rename = "company_fax")] pub fax: Option<String>,
```
Comment (TeXitoi) [inline src/objects.rs]: fax? really? :-P

Comment (prhod) [reply src/objects.rs]: removed in the navitia PR on NTFS

File: src/objects.rs [hunk 3]
```diff
@@ -345,15 +351,13 @@ impl Id<StopArea> for StopPoint {
 }
 impl_codes!(StopPoint);
 
-pub type Date = String;
```
Comment (TeXitoi) [inline src/objects.rs]: Better to keep this Date type and use it everywhere, it's simpler if we want to change.

Thus, something like `pub type Date = chrono::NaiveDate;`

Comment (datanel) [reply src/objects.rs]: fixed

**Pull request merged**